pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -413,8 +413,13 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 		delete(c.servicesMap, svcConv.Hostname)
 		delete(c.nodeSelectorsForServices, svcConv.Hostname)
 		delete(c.externalNameSvcInstanceMap, svcConv.Hostname)
+		_, isNetworkGateway := c.networkGateways[svcConv.Hostname]
 		delete(c.networkGateways, svcConv.Hostname)
 		c.Unlock()
+		if isNetworkGateway {
+			// networks are different, we need to update all eds endpoints
+			c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{Full: true, Reason: []model.TriggerReason{model.NetworksTrigger}})
```

Comment:
We did update xds in https://github.com/istio/istio/pull/33710/files#diff-e66bd631ae617f757133e4e6a817ab4a7964a71af09affb5f553a3772cbd1d81R464

Reply:
that is scoped to only the single service though so it won't be a full push of everything

Reply:
make sense